core/src/main/java/org/apache/iceberg/SnapshotManager.java

```diff
@@ -181,4 +182,12 @@ public void commit() {
       transaction.commitTransaction();
     }
   }
+
+  @Override
+  public void commitIf(List<Validation> validations) {
+    commitIfRefUpdatesExist();
+    // Add a no-op UpdateProperties to add given validations to transaction
+    transaction.updateProperties().commitIf(validations);
+    commit();
+  }
```

Comment:
`SnapshotManager` is the only `PendingUpdate` implementation where I have to implement the `commitIf` method "by hand" i.e. I can't just extend `BasePendingUpdate` like all the other implementations. This is because of the way `SnapshotManager` is implemented in terms of `Transaction` which means I don't have access to any base `TableMetadata` to `validate` directly. Instead, I add a conditional, no-op `UpdateProperties` to the underlying transaction which then  validates the current table state as part of the `Transaction` commit process.